src/utils/hooks/useYoutubeQuery.js

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useReducer } from 'react';
+import axios from 'axios';
+import { MAX_RESUTS_ON_SEARCH } from '../constants';
+
+const api = axios.create({
+  baseURL: 'https://www.googleapis.com/youtube/v3/',
+  params: {
+    part: 'snippet',
+    type: 'video',
+    order: 'relevance',
+    maxResults: MAX_RESUTS_ON_SEARCH,
+    key: process.env.REACT_APP_YOUTUBE_KEY,
+  },
+});
+
+// Reducer for hook state and actions
+const reducer = (state, action) => {
+  const { type, payload } = action;
+
+  switch (type) {
+    case 'idle':
+      return { status: 'idle', data: [], error: undefined };
+    case 'loading':
+      return { status: 'loading', data: [], error: undefined };
+    case 'success':
+      return { status: 'success', data: payload, error: undefined };
+    case 'error':
+      return { status: 'error', data: [], error: payload };
+    case 'pageToken':
```

Comment:
This case and error case return the same result could be use as
```suggestion
    case 'pageToken':
    case 'error':
        return { status: 'error', data: [], error: payload };
```